### PR TITLE
Join flow gender custom focus

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Step towards resolving #3053

### Changes:

* gets ref to custom gender input
* if custom gender option is clicked on, focuses cursor on input field
* corrects incorrect `htmlFor` value in `formik-radio-button` pointed out by @paulkaplan in https://github.com/LLK/scratch-www/pull/3178#discussion_r310618224

### Test Coverage:

n/a

### CODE REVIEW

File: src/components/formik-forms/formik-radio-button.jsx [hunk 1]
```diff
@@ -10,24 +10,26 @@ require('./formik-radio-button.scss');
 require('../forms/row.scss');
 
 const FormikRadioButtonSubComponent = ({
-    buttonValue,
```
Comment (picklesrus) [inline src/components/formik-forms/formik-radio-button.jsx]: why did you change the name of buttonValue?  Also, I'm confused again, what's the difference between field.value and value?

Comment (benjiwheeler) [reply src/components/formik-forms/formik-radio-button.jsx]: Yeah, this is one of the most confusing things about formik's handling of radio buttons...

So, when you give the formik `<Field>` component a `subComponent` prop (which you need to do for radio buttons because there's no built-in handling of them, unlike inputs and selects... which is annoying), it automatically passes it a `field` prop. And what's weird is, `field.value` is the value of the _radio button group_, not this individual button. I can't figure out why this is, but that's the `field.value` when I inspect/log it.

Meanwhile, `value` is the value of this individual button. 

So to see if this button should be currently "selected", you need to compare those two kinds of value.

Comment (benjiwheeler) [reply src/components/formik-forms/formik-radio-button.jsx]: oh, and I thought that `value` would be simpler than `buttonValue`. What do you think?

Comment (picklesrus) [reply src/components/formik-forms/formik-radio-button.jsx]: I see (maybe :) ). In this example, are they using a strategic value of id to get around this? https://codesandbox.io/s/pjqp3xxq7q

Comment (benjiwheeler) [reply src/components/formik-forms/formik-radio-button.jsx]: In that example, I think this section of `CheckboxGroup` is doing a lot of work:

```
            return React.cloneElement(child, {
              field: {
                value: value.includes(child.props.id),
                onChange: this.handleChange,
                onBlur: this.handleBlur
              }
            });
```

Comment (picklesrus) [reply src/components/formik-forms/formik-radio-button.jsx]: Sorry, I meant the RadioGroup. In the UI (and I think the code too) the CheckboxGroup is a separate thing.  Sorry to to get stuck on this point. Just want to make sure I have a better understanding of how it works.

Comment (benjiwheeler) [reply src/components/formik-forms/formik-radio-button.jsx]: Sorry, yes, your previous comment is exactly right. In that codesandbox example, they use the `id` prop to set `value`, and they use `field.value` to compare to `id` to determine if the button is selected.

Make sense? I guess formik is acting like all the radio buttons are a single "field", or something? I don't totally understand it, to tell the truth.
